moveit_core/robot_state/test/robot_state_test.cpp

```diff
@@ -553,6 +554,92 @@ TEST(FK, OneRobot)
   EXPECT_NEAR_TRACED(state.getGlobalLinkTransform("link_e").translation(), Eigen::Vector3d(2.8, 0.6, 0));
 }
 
+class LoadPR2 : public testing::Test
+{
+protected:
+  virtual void SetUp()
+  {
+    std::string res_path(MOVEIT_TEST_RESOURCES_DIR);
+
+    urdf_model = urdf::parseURDFFile(res_path + "/pr2_description/urdf/robot.xml");
+    srdf_model.reset(new srdf::Model());
+    srdf_model->initFile(*urdf_model, res_path + "/pr2_description/srdf/robot.xml");
+    robot_model.reset(new moveit::core::RobotModel(urdf_model, srdf_model));
+  }
+
+  virtual void TearDown()
+  {
+  }
+
+protected:
+  urdf::ModelInterfaceSharedPtr urdf_model;
+  srdf::ModelSharedPtr srdf_model;
+  moveit::core::RobotModelConstPtr robot_model;
+};
+
+void generateTestTraj(std::vector<std::shared_ptr<robot_state::RobotState>>& traj,
+                      const moveit::core::RobotModelConstPtr& robot_model,
+                      const robot_model::JointModelGroup* joint_model_group)
+{
+  traj.clear();
+
+  // 3 waypoints with default joints
+  std::shared_ptr<robot_state::RobotState> robot_state(new robot_state::RobotState(robot_model));
+  robot_state->setToDefaultValues();
+  for (std::size_t traj_ix = 0; traj_ix < 3; ++traj_ix)
+    traj.push_back(robot_state);
+
+  // 4th waypoint with a large jump of 1.01 in first joint
+  robot_state.reset(new robot_state::RobotState(*robot_state));
+  std::vector<double> joint_positions;
+  robot_state->copyJointGroupPositions(joint_model_group, joint_positions);
+  joint_positions[0] -= 1.01;
+  robot_state->setJointGroupPositions(joint_model_group, joint_positions);
+  traj.push_back(robot_state);
```

Comment:
On further reflection, I think we should have the cutoff further than 1 index from the length of the list. Returning 3/4 is susceptible to off by one errors

Reply:
I don't get this point. Can you please elaborate?

Reply:
Off by one errors are one of the most common types of bugs. The probability that some future bug slips past the test because of an off by one error that returns the length of the input trajectory minus one rather than actually finding the jump is non-trivial. While developing #821 I found a bug in my own code that did exactly this. It would be really easy to add another point to the trajectory so that the test would expect 3 out of 5 rather than 3 out of 4. 

This was the thinking behind the changes that got reverted in 20dc432.